addons/binding/org.openhab.binding.mysensors/src/main/java/org/openhab/binding/mysensors/handler/MySensorsThingHandler.java

```diff
@@ -0,0 +1,396 @@
+
+import static org.openhab.binding.mysensors.MySensorsBindingConstants.*;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Map;
+
+import org.eclipse.smarthome.core.library.types.DateTimeType;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.ThingStatusInfo;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.State;
+import org.openhab.binding.mysensors.MySensorsBindingConstants;
+import org.openhab.binding.mysensors.config.MySensorsSensorConfiguration;
+import org.openhab.binding.mysensors.converter.MySensorsTypeConverter;
+import org.openhab.binding.mysensors.internal.event.MySensorsGatewayEventListener;
+import org.openhab.binding.mysensors.internal.event.MySensorsNodeUpdateEventType;
+import org.openhab.binding.mysensors.internal.gateway.MySensorsGateway;
+import org.openhab.binding.mysensors.internal.protocol.MySensorsAbstractConnection;
+import org.openhab.binding.mysensors.internal.protocol.message.MySensorsMessage;
+import org.openhab.binding.mysensors.internal.sensors.MySensorsChild;
+import org.openhab.binding.mysensors.internal.sensors.MySensorsChildConfig;
+import org.openhab.binding.mysensors.internal.sensors.MySensorsNode;
+import org.openhab.binding.mysensors.internal.sensors.MySensorsNodeConfig;
+import org.openhab.binding.mysensors.internal.sensors.MySensorsVariable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link MySensorsThingHandler} is responsible for handling commands, which are
+ * sent to one of the channels and messages received via the MySensors network.
+ *
+ * @author Tim Oberföll
+ */
+public class MySensorsThingHandler extends BaseThingHandler implements MySensorsGatewayEventListener {
+
+    private Logger logger = LoggerFactory.getLogger(getClass());
+
+    private MySensorsSensorConfiguration configuration = null;
+
+    private int nodeId = 0;
+    private int childId = 0;
+    private boolean requestAck = false;
+    private boolean revertState = true;
+
+    private boolean smartSleep = false;
+
+    private int expectUpdateTimeout = -1;
+
+    private DateTimeType lastUpdate = null;
+
+    private MySensorsGateway myGateway;
+
+    public MySensorsThingHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        configuration = getConfigAs(MySensorsSensorConfiguration.class);
+        nodeId = Integer.parseInt(configuration.nodeId);

```

Comment:
Why don't you store them as an Integer then? Then you don't need to convert them here. Just keep the reference to the configuration and you would be done.